File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int[] arr = new int[n];
        for(int i = 0; i < n; i++) {
            arr[i] = i+1;
        }
        int lt = 0, count = 0, sum = 0;
        for(int rt = 0; rt < n; rt++) {
            sum += arr[rt];
            if(sum == n) {
                count++;
            }
            while(n <= sum) {
                sum -= arr[lt++];
                if(sum == n) {
                    count++;
                }
            }
        }
        return count;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
const solution = (n) => {
    let count = 0;
    
    for(let i = 1; i * (i - 1) / 2 < n; i++) {
        const num = (n - i * (i - 1) / 2) / i;
        
        if(!(num % 1)) count++;
    }
    
    return count;
}